node/src/accountant/payable_dao.rs

```diff
@@ -582,35 +745,163 @@ mod tests {
             .unwrap();
         let subject = PayableDaoReal::new(conn);
 
-        let _ = subject.mark_pending_payable_rowid(&wallet, rowid);
+        let _ = subject.mark_pending_payables_rowids(&[(&wallet, rowid)]);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Marking pending payable rowid for wallets 0x000000000000000000000000000000686f6f6761, \
+         0x000000000000000000000000000000626f6f6761, 0x00000000000000000000626f6f6761686f6f6761 affected 1 rows but expected 3. \
+         Accounts for wallets (0x000000000000000000000000000000626f6f6761, 0x00000000000000000000626f6f6761686f6f6761) \
+         must have contained some populated pending payable rowids during the last write of the newest ids down to the respective \
+         table. System unreliable. Impaired, double payments may be suspected the cause in a situation like this"
+    )]
+    fn mark_pending_payables_rowids_refuses_to_overwrite_existing_marked_rowids() {
+        let home_dir = ensure_node_home_directory_exists(
+            "payable_dao",
+            "mark_pending_payables_rowids_refuses_to_overwrite_existing_marked_rowids",
+        );
+        let wallet_1 = make_wallet("hooga");
+        let rowid_1 = 550;
+        let wallet_2 = make_wallet("booga");
+        let rowid_2 = 555;
+        let wallet_3 = make_wallet("boogahooga");
+        let rowid_3 = 558;
+        let conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        insert_payable_record_fn(&*conn, &wallet_1.to_string(), 12345, 1_000_000_000, None);
```

Comment:
You said something here about iterating over an array, and thought it might be more concise.